vms/subnet.go

```diff
@@ -0,0 +1,257 @@
+package vms
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/ava-labs/avalanche-network-runner/network"
+
+	"github.com/ava-labs/avalanchego/api"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+)
+
+// CustomVM wraps data to create a custom VM
+type CustomVM struct {
+	Path     string
+	Genesis  string
+	Name     string
+	SubnetID string
+	ID       string
+}
+
+const (
+	loopTimeout         = 1 * time.Second
+	longTimeout         = 10 * loopTimeout
+	defaultKeyThreshold = 1
+
+	validatorWeight    = 3000
+	validatorStartDiff = 30 * time.Second
+	validatorEndDiff   = 30 * 24 * time.Hour // 30 days
+)
+
+// SetupSubnet creates the necessary transactions to create a subnet for a given custom VM.
+// It then also waits until all these transactions are confirmed on all nodes.
+// Finally it creates a blockchain and makes sure all nodes of the given network
+// are validating this blockchain.
+// It requires a `privateKey` in order to issue the necessary transactions
+func SetupSubnet(
```

Comment:
should this be part of the network API?

Reply:
It could. I thought it to have it very separate first.